devito/propagator.py

```diff
@@ -178,8 +212,17 @@ def generate_space_loops_blocking(self, loop_body):
         ivdep = True
         remainder = False
         orig_loop_body = loop_body
+
+        inner_most_dim_passed = False  # used in oder to avoid at inner most dimension
+
         for spc_var, block_size in reversed(zip(list(self.space_dims), self.block_sizes)):
             dim_var = str(self._var_map[spc_var])
+
+            if self.auto_tune and inner_most_dim_passed:  # change block size into var
```

Comment:
I dont think this can be avoided, unless we write a function within c code that somehow adjusts the loop counts based on block size that is beeing at-ed.
However according to @pvelesko the variation in time shouldnt be more that 10%. So I thought its acceptable for now. As the performance figure fluctuate up to 50% based on block sizes